crates/sui-bytecode-src-verifier/src/lib.rs

```diff
@@ -0,0 +1,253 @@
+                        pkg_symbol, None,
+                    ));
+                }
+            };
+
+            for (symbol, addr) in resolution_package.resolution_table {
+                // package addresses may show up many times, but we only need to verify them once
+                // zero address is the package we're checking dependencies for
+                if verified_dependencies.contains_key(&addr) || addr.eq(&AccountAddress::ZERO) {
+                    continue;
+                }
+
+                // fetch the Sui object at the address specified for the package in the local resolution table
+                let on_chain_package = self.pkg_for_address(&addr).await?;
+
+                for (oc_name, on_chain_bytes) in &on_chain_package.module_map {
+                    let on_chain_module_symbol = Symbol::from(oc_name.as_str());
+                    let local_bytes = match local_pkg_bytes.get(&on_chain_module_symbol) {
+                        Some(bytes) => bytes,
+                        None => {
+                            return Err(DependencyVerificationError::LocalDependencyNotFound(
+                                pkg_symbol,
+                                Some(on_chain_module_symbol),
+                            ))
+                        }
+                    };
+
+                    // compare local bytecode to on-chain bytecode to ensure integrity of our dependencies
+                    if local_bytes != on_chain_bytes {
+                        return Err(DependencyVerificationError::ModuleBytecodeMismatch(
+                            pkg_symbol.to_string(),
+                            on_chain_module_symbol.to_string(),
+                            addr,
+                            local_bytes.clone(),
+                            on_chain_bytes.clone(),
+                        ));
+                    }
+
+                    if self.verbose {
+                        println!(
+                            "{}::{} - {} bytes, code matches",
+                            pkg_symbol,
+                            oc_name,
+                            on_chain_bytes.len()
+                        );
+                    }
+                }
+
+                on_chain_module_count += on_chain_package.module_map.len();
+
+                verified_dependencies.insert(
+                    addr,
+                    Dependency {
+                        symbol: symbol.to_string(),
+                        module_bytes: on_chain_package.module_map.clone(),
+                    },
+                );
+            }
+        }
+
+        // total number of modules in packages must match, in addition to each individual module matching
+        if compiled_package.deps_compiled_units.len() != on_chain_module_count {
+            let len = compiled_package.deps_compiled_units.len();
+            return Err(DependencyVerificationError::ModuleCountMismatch(
```

Comment:
I think here, it would be helpful to say whether which module(s) are missing

Reply:
Added a `Vec<String>` (where the strings are `package_name::module_name`) containing all the missing modules to this